doc/designs/application-specific-passwords.md

```diff
@@ -0,0 +1,283 @@
+# Application-specific passwords
+
+**DESIGN STAGE**
+
+## Overview
+
+FreeIPA does not currently allow users to have more than one password.
+Using a single password for the user's account carries potential risk and
+inconvenience with it. If the password is compromised on only one of the devices
+the user uses, the attacker gains full administrative access to the user's
+account and all services associated with it. The user can then halt it only by
+changing this single account's password and reconfiguring all systems and
+applications using that password for authentication (even if only to one
+particular service) to use the new one.
+
+This enhancement therefore brings support multiple passwords for a single user
+and extends the HBAC access control to allow specifying what application-specific
```

Comment:
It would be great to unroll the `HBAC` abbreviation once in this document, ideally here at the first mention.

Googling it brings up `Home Birth After Cesaerian` first and then `Host-Based` and `History-Based` access control, which both fall into this domain, but I wouldn't know which is meant 🙏

Reply:
All right, I rewrote it to 

> ...extends the HBAC (host-based access control) to allow...

It should be clear now.

Reply:
@dimaqq in FreeIPA context all these terms have quite concrete meaning. The document is not supposed to be used outside of the FreeIPA context, so some of the terms are assumed to be known.
May be we can have a vocabulary of terms for the whole project but in general familiarity with project essentials is assumed.